pytorch_lightning/core/lightning.py

```diff
@@ -1969,3 +1969,188 @@ def _register_sharded_tensor_state_dict_hooks_if_available(self) -> None:
 
         self._register_state_dict_hook(state_dict_hook)
         self._register_load_state_dict_pre_hook(pre_load_state_dict_hook, True)
+
+    def _init_optimizers_and_lr_schedulers(self) -> Tuple[List, List, List]:
```

Comment:
on second thought, I think @awaelchli 's suggestion of moving this to a pure utilities function/file is much better. That avoids a ton of bloat in the LightningModule API.
```
init_optimizers_and_lr_schedulers(lightning_module: LightningModule) -> ???:
   ...
```

then we can also add nice unit tests for this to cover all of the return type options

Reply:
I would only make it a method if we expect anyone to override the parsing, but at this point I don't think this is a good idea. Function is much better for unit testing as pointed out by @ananthsub. Regardless, I am still in favor of keeping the function in core, arguing that this is where the hook and return type is defined in the first place, and fighting against the current behavior of using the pl/utilities as a dumping ground for random functions :)

Reply:
I think this makes sense to add as a set of utility functions under core/optimizers then

Reply:
> I think this makes sense to add as a set of utility functions under core/optimizers then

Done!